src/ts/richtext/menu.ts

```diff
@@ -0,0 +1,85 @@
+import { toggleMark } from "prosemirror-commands";
+import { MenuElement, MenuItem, MenuItemSpec } from "prosemirror-menu";
+import { MarkType, NodeType, Schema } from "prosemirror-model";
+import { Command, EditorState } from "prosemirror-state";
+import { isInNode, toggleBulletListCommand } from "./utils/listsHelpers";
+import { linkItemCommand, unlinkItemCommand } from "./utils/command-helpers";
+
+export const cmdItem = (cmd: Command, options: Partial<MenuItemSpec>) => {
```

Comment:
To avoid having to cast options.title as string:
```suggestion
export const cmdItem = (cmd: Command, options: Partial<MenuItemSpec>  & { title: string }) => {
```
(Applies multiple times in this file)

Reply:
Delightful, implemented now.